main/src/ca/uwaterloo/flix/language/phase/EntryPoints.scala

```diff
@@ -106,6 +107,43 @@ object EntryPoints {
     }
   }
 
+  /**
+    * Returns a new shell function that swaps the unit value for a call to _f.
+    *
+    * This is necessary because the expression being evaluated may have non primitive effects
+    * with default handlers.
+    *
+    * The set of effects with default handlers is not known from the start unlike the set of primitive effects.
+    * In order to support the current way of creating a shell (create some simple wrapping code and execute it),
+    * we need to substitute PrimitiveEffs with the real set of effects generated by exp at EntryPoints.
+    *
+    * Takes
+    * {{{
+    *   def main(): Unit \ PrimitiveEffs =
+    *     def _f()={ println(exp) }
+    *     checked_ecast(())
+    * }}}
+    *
+    * Returns
+    * {{{
+    *   def main(): Unit \ expEffs + IO =
+    *     def _f()={ println(exp) }
+    *     _f()
+    * }}}
```

Comment:
I think I understand better now. Maybe you want to start the documentation by saying we have this code snippet and we want to transform it to this other snippet.